File: week2/Ariling/prgs-방문길이/main.py
def solution(dirs):
    # x축 -5 ~ 5 y축 -5 ~ 5
    dir_set = set()
    x, y = 0, 0
    for dir in dirs:
        newX, newY = x, y
        if dir == "U" and y < 5:
            newY += 1
        elif dir == "D" and y > -5:
            newY -= 1
        elif dir == "L" and x > -5:
            newX -= 1
        elif dir == "R" and x < 5:
            newX += 1
        if newX == x and newY == y:
            continue
        else:
            dir_set.add((y, x, newY, newX))
            dir_set.add((newY, newX, y, x))
            x,y = newX, newY
    answer = len(dir_set) // 2
    return answer

print(solution("LULLLLLLU"))

File: week2/Ariling/prgs-방문길이/main.js
function solution(dirs) {
  var answer = 0;
  var move_set = new Set();
  var x = 0;
  var y = 0;
  for (let i = 0; i < dirs.length; i++) {
    var newX = x;
    var newY = y;
    if (dirs[i] === "U" && y + 1 <= 5) {
      newY += 1;
    } else if (dirs[i] === "D" && y - 1 >= -5) {
      newY -= 1;
    } else if (dirs[i] === "L" && x - 1 >= -5) {
      newX -= 1;
    } else if (dirs[i] === "R" && x + 1 <= 5) {
      newX += 1;
    }
    if (newX == x && newY == y) {
      continue;
    } else {
      move_set.add(`${y},${x},${newY},${newX}`);
      move_set.add(`${newY},${newX},${y},${x}`);
      x = newX;
      y = newY;
    }
  }
  answer = Math.floor(move_set.size / 2);
  return answer;
}
